### PR TITLE
[#445] Add /logout routing and clear storage on logout (inconsistent state of storage)

### LINKED ISSUE #445: Add /logout routing and clear storage on logout (inconsistent state of storage)
- [ ] Create LogoutComponent and add routing to it
- [ ] Navigate to login page
- [ ] Should notification info "You have been logged out"
- [ ] Repair storage state after login, logout and login actions

### PR DESCRIPTION
**Pull request by danielkryska**
- logout can be done by navigating to link `/logout`
- when log out button clicked, a user should be navigated to `/login` page
- Storage error (after login, logout, login and moving to `/settings/profile`) have been repaired

### CODE REVIEW

File: s4e-web/src/app/app.routes.ts [hunk 1]
```diff
@@ -11,6 +12,10 @@ export const appRoutes: Routes = [
     path: 'login',
     component: LoginComponent,
     canActivate: [IsNotLoggedIn]
+  }, {
+    path: 'logout',
+    component: LogoutComponent // ,
+    // canActivate: [IsNotLoggedIn]
```
Comment (michal-szostak) [inline s4e-web/src/app/app.routes.ts]: Don't you think that `cabActivate: [IsLoggedIn]` would be good?

File: s4e-web/src/app/views/logout/logout.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        LogoutModule,
+        RouterTestingModule,
+        HttpClientTestingModule
+      ],
+      providers: [
+        SessionService,
```
Comment (michal-szostak) [inline s4e-web/src/app/views/logout/logout.component.spec.ts:22]: Is it necessart? If so then this dependency should be moved to `LogoutModule`

File: .gitignore [hunk 3]
```diff
@@ -41,3 +41,4 @@ testem.log
 .DS_Store
 Thumbs.db
 /tmp
+/s4e-web/.vscode
```
Comment (jswk) [inline .gitignore:44]: I'd rather edit the section, which is explicitly about VSCode, lines 14-18. It assumes you want to commit some settings and so on, and you probably don't want to do this.

File: s4e-web/src/app/state/session/session.service.ts [hunk 4]
```diff
@@ -60,8 +64,19 @@ export class SessionService {
 
   @action('logout')
   logout() {
-    this.setToken(null, null);
-    resetStores();
-    this.router.navigate(['/login']);
+    if (this.sessionQuery.isLoggedIn()) {
+      console.log("JESTEM");
```
Comment (jswk) [inline s4e-web/src/app/state/session/session.service.ts:68]: ?

File: s4e-web/src/app/state/session/session.service.ts [hunk 5]
```diff
@@ -60,8 +64,19 @@ export class SessionService {
...
+      this.setToken(null, null);
+
+      const stores = persistState();
+      stores.clearStore('Notification');
+
+      akitaConfig({
+        resettable: true
+      });
```
Comment (jswk) [inline s4e-web/src/app/state/session/session.service.ts:76]: Why is this needed? Shouldn't config be done in some initialization code, and not in logout function?

**Comment by kmarszalek** [comment]
@wziajka 502 Bad Gateway on config. https://pr-446-sok.docker-fid.grid.cyf-kr.edu.pl/ :(

**Comment by kmarszalek** [comment]
could you provide me with some screenshots of notification after logout? (chrome)
logout works fine but i dont see notification

changelog order. new entries on top.

File: CHANGELOG.md [hunk 6]
```diff
@@ -10,6 +10,7 @@ The most recent changes are on top, in each type of changes category.
 ### Added
 
 - Audit selected entities [#158](https://github.com/cyfronet-fid/sat4envi/issues/158)
+- Add /logout routing and clear storage on logout (inconsistent state of storage) [#445](https://github.com/cyfronet-fid/sat4envi/issues/445)
```
Comment (kmarszalek) [inline CHANGELOG.md:13]: changelog order. new entries on top.

File: .gitignore (hunk 3 above)
Comment (jswk) [reply .gitignore]: I was thinking more like leaving just this in the VSCode section:
```
# IDE - VSCode
.vscode/*
```
as I don't think anyone else depends on it in our team, and it would be cleaner.

**Comment by jswk** [comment]
> @wziajka 502 Bad Gateway on config. https://pr-446-sok.docker-fid.grid.cyf-kr.edu.pl/ :(

It works now.

File: .gitignore (hunk 3 above)
Comment (danielkryska) [reply .gitignore]: ok

File: CHANGELOG.md (hunk 6 above)
Comment (danielkryska) [reply CHANGELOG.md:13]: Is it influencing anything? Don't we deploy things in packages of features? Does it matter for the client, what was an order of merges?

**Comment by danielkryska** [comment]
I'm not surprised that notification doesn't appear..., it has been removed due to redundant information (user will know that he's/she's logged out, because he/she triggered logout button).
